tests/integration/targets/cloud_discovery_providers_gcp/tasks/main.yml

```
@@ -0,0 +1,510 @@
+---
+- module_defaults:
+    group/infoblox.universal_ddi.all:
+      portal_url: "{{ portal_url }}"
+      portal_key: "{{ portal_key }}"
+
+  block:
+    # Create a cloud discovery with a random name to avoid conflicts
+    - ansible.builtin.set_fact:
+        cloud_discovery_provider_gcp_name: "test_gcp_{{ 999999 | random | string }}"
+        cloud_discovery_provider_gcp_name_auto: "test_gcp_auto{{ 999999 | random | string }}"
+        config_access_id: "1{{ 999999999999 | random | string }}"
+        config_access_id_auto: "1{{ 999999999999 | random | string }}"
+
+    - name: Create a GCP cloud discovery provider (check mode)
+      infoblox.universal_ddi.cloud_discovery_providers:
+        name: "{{ cloud_discovery_provider_gcp_name }}"
+        provider_type: "Google Cloud Platform"
+        account_preference: "single"
+        credential_preference:
+          access_identifier_type: "project_id"
+          credential_type: "dynamic"
+        source_configs:
+          - credential_config:
+                access_identifier: "{{ config_access_id }}"
+            restricted_to_accounts:
+              - "{{ config_access_id }}"
+        state: present
+      check_mode: yes
+      register: gcp_provider
+    - name: Get Infoblox GCP cloud discovery provider
+      infoblox.universal_ddi.cloud_discovery_providers_info:
+        filters:
+          name: "{{ cloud_discovery_provider_gcp_name }}"
+      register: gcp_provider_info
+    - assert:
+        that:
+          - gcp_provider is changed
+          - gcp_provider_info is not failed
+          - gcp_provider_info.objects | length == 0
+
+    - name: Create an GCP cloud discovery provider
```

```suggestion
    - name: Create a GCP cloud discovery provider
```
Please apply these changes for all other task names